janus_server/src/datastore.rs

```diff
@@ -615,6 +620,67 @@ impl<C: Clock> Transaction<'_, C> {
             .collect::<Result<Vec<Nonce>, Error>>()
     }
 
+    /// `get_unaggregated_client_report_nonces_by_collect_for_task` returns pairs of nonces and
+    /// aggregation parameters, corresponding to client reports that have not yet been aggregated,
+    /// or not aggregated with a certain aggregation parameter, and for which there are collect
+    /// jobs, for a given task.
+    ///
+    /// This should only be used with VDAFs with a non-unit type aggregation parameter. If a VDAF
+    /// has the unit type as its aggregation parameter, then
+    /// `get_unaggregated_client_report_nonces_for_task` should be used instead. In such cases, it
+    /// is not necessary to wait for a collect job to arrive before preparing reports.
+    #[cfg(test)]
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_unaggregated_client_report_nonces_by_collect_for_task<const L: usize, A>(
+        &self,
+        task_id: TaskId,
+    ) -> Result<Vec<(Nonce, A::AggregationParam)>, Error>
+    where
+        A: vdaf::Aggregator<L>,
```

Comment:
```suggestion
        A: vdaf::Aggregator<L> + VdafHasAggregationParameter,
```

Now that we have the marker trait, could we use it here (and `()` in `get_unaggregated_client_report_nonces_for_task`) for a bit more safety?

Reply:
`get_unaggregated_client_report_nonces_for_task()` does not have any type parameters currently, so I'd be a bit loath to introduce one just for the constraint.